hw/ip_templates/racl_ctrl/rtl/racl_ctrl.sv.tpl

```diff
@@ -181,31 +181,38 @@ module ${module_instance_name} import ${module_instance_name}_reg_pkg::*; #(
   logic first_error;
   assign first_error = ~reg2hw.error_log.valid.q & racl_error_arb.valid;
 
-  // Writing 1 to the error valid bit clears the log and log address again
-  logic clear_log;
-  assign clear_log = reg2hw.error_log.valid.q & reg2hw.error_log.valid.qe;
-
-  assign hw2reg.error_log.valid.d  = ~clear_log;
-  assign hw2reg.error_log.valid.de = racl_error_arb.valid | clear_log;
-
-  // Overflow is raised when error is valid and a new error is coming in or more than one
-  // error is coming in at the same time
-  assign hw2reg.error_log.overflow.d  = ~clear_log;
-  assign hw2reg.error_log.overflow.de = (reg2hw.error_log.valid.q & racl_error_arb.valid) |
-                                        racl_error_arb.overflow                           |
-                                        clear_log;
-
-  assign hw2reg.error_log.read_access.d  = clear_log ? '0 : racl_error_arb.read_access;
-  assign hw2reg.error_log.read_access.de = first_error | clear_log;
-
-  assign hw2reg.error_log.role.d  = clear_log ? '0 : racl_error_arb.racl_role;
-  assign hw2reg.error_log.role.de = first_error | clear_log;
-
-  assign hw2reg.error_log.ctn_uid.d  = clear_log ? '0 : racl_error_arb.ctn_uid;
-  assign hw2reg.error_log.ctn_uid.de = first_error | clear_log;
-
-  assign hw2reg.error_log_address.d  = clear_log ? '0 : racl_error_arb.request_address;
-  assign hw2reg.error_log_address.de = first_error | clear_log;
+  assign hw2reg.error_log.valid.d  = 1'b1;
+  assign hw2reg.error_log.valid.de = racl_error_arb.valid;
+
+  // The register's overflow signal should go high if:
+  //
+  // - The overflow signal from racl_error_arb is true (which means that the arbiter saw more than
+  //   one error reported at once)
+  //
+  // - The register's valid signal is already true but racl_error_arb.valid is true. In this case,
+  //   the overflow is accumulating over time.
+  //
+  // It should be cleared again when we write to the register (clearing the valid signal). To
+  // simplify the logic, allow it to drop one cycle later.
+
+  assign hw2reg.error_log.overflow.d  = racl_error_arb.overflow |
+                                        (reg2hw.error_log.valid.q & racl_error_arb.valid);
+
+  assign hw2reg.error_log.overflow.de = racl_error_arb.overflow |
+                                        (reg2hw.error_log.valid.q & racl_error_arb.valid) |
+                                        !reg2hw.error_log.valid.q;
+
+  assign hw2reg.error_log.read_access.d  = racl_error_arb.read_access;
+  assign hw2reg.error_log.read_access.de = first_error;
```

Comment:
Is the value in that register being kept when valid is cleared via the `rw1c` behavior?

Reply:
Good point: this isn't consistent. At the moment, we clear `overflow` (a cycle late) but keep the `read_access` flag until the first new error appears. I'd probably be inclined to clear this as well?

Reply:
Do you mean that the fields of the register have a meaning when `valid = 0`? If not, maybe it makes sense to just leave them with their values from last time around? But I guess there's a potential information disclosure, so we could just zero everything?

Reply:
One issue for the oscillation is I think that qe gets asserted for software AND hardware writes.

@andreaskurth i think this one of those cases as well.

Reply:
Yep, that's exactly what was causing the problem. I'll push a version in a second that wipes all the other fields as well.